### ISSUE TITLE
can you provide contribution guide?

### ISSUE THREAD

**Issue by andrzejsliwa** [singer-io/tap-github#98]
- especially info about how to setup dev env
- how to execute tests
- how to make integrations tests running 
- etc...

**Comment by raulgzm** [comment]
Yeah, I agree with this request. And maybe it would be great if we could contemplate the possibility to refactor the structure of the tap in order to have a better design that facilitates contributions and evolution of the tap.
